src/lib/promise-poller.js

```diff
@@ -1,35 +1,67 @@
-import Promise from 'bluebird';
-import debugModule from 'debug';
-import strategies from './strategies';
+import Promise from "bluebird";
+import debugModule from "debug";
+import strategies from "./strategies";
 
-const debug = debugModule('promisePoller');
+const debug = debugModule("promisePoller");
 
 const DEFAULTS = {
-  strategy: 'fixed-interval',
+  strategy: "fixed-interval",
   retries: 5,
-  shouldContinue: (err) => {
+  shouldContinue: err => {
     return !!err;
   }
 };
 
 let pollerCount = 0;
 
+/**
+ *
+ * 
+ * @export
+ * @param {Object} options that contains polling options
+ * @param {function} options.taskFn a function that must resolve or reject a promise or return false to stop polling.
+ * @param {Number} options.interval number of ms until taskFn runs again.
+ * @param {Number} [options.retries=5] number of times taskFn is attempted.
+ * @param {String} [options.strategy='fixed-interval'] one of 'fixed-interval', 'linear-backoff', 'exponential-backoff'.
+ * @param {Number} [options.masterTimeout=null] ms to reject taskFns promise regardless of retries.
+ * @returns {Promise} returns taskFn's resolved promise or array of rejections
+ * @example 
+ * promisePoller({
+ *   taskFn: yourFn(),
```

Comment:
It was just this line that was bad, the rest of the doc was good, IMO.

Reply:
Yup you are completely right that is bad should I redo it or is it not worth it.